portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,20 +45,30 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     int commentLimit = Integer.parseInt(request.getParameter("limit"));
     String languageCode = request.getParameter("languageCode");
-    System.out.println(languageCode);
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<Entity> queryList = results.asList(FetchOptions.Builder.withLimit(commentLimit));
 
+    Translate translate = null;
+    if (!languageCode.equals("original")) {
+      translate = TranslateOptions.getDefaultInstance().getService();
+    }
+
     ArrayList<Comment> commentList = new ArrayList<Comment>();
     for (Entity entity : queryList) {
       long id = entity.getKey().getId();
       String name = (String) entity.getProperty("name");
       String comment = (String) entity.getProperty("comment");
       long timestamp = (long) entity.getProperty("timestamp");
 
+      // Do the translation.
+      if (!languageCode.equals("original")) {
+        Translation translation = translate.translate(comment, Translate.TranslateOption.targetLanguage(languageCode));
```

Comment:
Do you want to add error handling for TranslateException?

Reply:
I added a try catch block which handles error handling for TranslateException. In the catch block I made a println statement which prints the error in the terminal, but should the user be able to see that the error occurred? I was thinking about making a variable called error inside the comment class to be displayed using javascript when the json is returned. What do you think about that?

Reply:
So I made a commit which catches the error and sends it back to script.js as a response which script.js would then show as an alert.

Reply:
Usually user see some error message to know there is an error but without detailed error message which is related to detailed server implementation. Server could have server logs for debugging on server side with more detailed info.